docs/design/operator.md

```diff
@@ -159,6 +161,7 @@ spec:
       priorityClassName: system-cluster-critical
       labels:
         app: provisioner
+      privileged: true  
```

Comment:
why `privileged`? As it has unrestricted perms and logrotate only needs (afaiu) access to hostPath, why not `hostaccess`?

I suppose the implementation of API is in rook/csi, could you pls point to it (or) which one of below is the implementation?
- are we using kubelet level logrotation
- linux logrotate and we translate what mentioned here to logrotate config
- custom golang/bash rotation

thanks.

Reply:
>linux logrotate and we translate what mentioned here to logrotate config

Reply:
>why privileged? As it has unrestricted perms and logrotate only needs (afaiu) access to hostPath, why not hostaccess?

As this is how open shift defines it https://docs.openshift.com/container-platform/4.10/storage/persistent_storage/persistent-storage-hostpath.html#persistent-storage-hostpath-pod_persistent-storage-hostpath

Reply:
ack, I am thinking this from ServiceAccount perspective, if sa has corresponding scc, will that not be enough https://docs.openshift.com/container-platform/4.16/authentication/managing-security-context-constraints.html?

I believe securityContext isn't specific to OCP but SCC is and so you mean to have both securityContext and SCC specified? Maybe we only need `allowPrivilegeEscalation` in SC and `hostaccess` in SCC?

Reply:
I was thinking the same, but we specially need the privileged to mount the directory.
More discussion https://github.com/rook/rook/issues/14400